client/rpc-servers/src/lib.rs

```diff
@@ -104,9 +109,14 @@ pub fn start_ws<M: Send + Sync + 'static>(
 	let mut builder = WsServerBuilder::new()
 		.max_request_body_size(max_request_body_size as u32)
 		.max_connections(max_connections as u64)
-		.set_id_provider(RandomStringIdProvider::new(16))
 		.custom_tokio_runtime(rt.clone());
 
+	if let Some(provider) = id_provider {
```

Comment:
did this to avoid to have to `Box<RandomStringIdProvider>`, a little awkward but...